### PR TITLE
added open timeout to http client

### PR DESCRIPTION
**Pull request by benoist**
I've added the open timeout. Not sure how to test the unhappy flow. If it's required let me know how to test it and I will add it :)

Also continuing from the discussion in my previous PR, I'm not sure about the write timeout and when that would occur.  Also couldn't find it in the Ruby Net::HTTP client, not that we should really lead from that example though ;)

### CODE REVIEW

File: src/http/client/client.cr [hunk 1]
```diff
@@ -114,6 +114,16 @@ class HTTP::Client
     self.read_timeout = read_timeout.total_seconds
   end
 
+  # Set the number of seconds to wait when opening before raising an `IO::Timeout`.
+  def open_timeout=(open_timeout : Number)
+    @open_timeout = open_timeout.to_f
+  end
+
+  # Set the open timeout with a Time::Span, to wait when opening before raising an `IO::Timeout`.
```
Comment (jhass) [inline src/http/client/client.cr]: I think an example would be great here, also link `Time::Span` with backticks.

```
client = HTTP::Client.new("example.org")
client.open_timeout = 5.minutes
begin
  response = client.get("/")
rescue IO::Timeout
  puts "Timeout!"
end
```

**Comment by benoist** [comment]
I've added the documentation for the open and read timeouts.

@asterite Would it also be wise to set some default values so that the default implementation is guarded against hanging connections?

**Comment by benoist** [comment]
I've renamed open_timeout to connect_timeout as this fits the error message better

**Comment by jhass** [comment]
:+1: LGTM

**Comment by jhass** [comment]
I guess nobody objects, thanks!

**Comment by Nakilon** [comment]
Sorry for being late, but it is named `open_timeout` in Ruby and I think that error message calls it in a slightly another way isn't a big deal.

**Comment by jhass** [comment]
We called the TCPSocket option `connect_timeout` too, so it's good to be consistent with that.

**Pull request merged**